pkg/storage/saver.go

```diff
@@ -2,10 +2,9 @@ package storage
 
 import (
 	"context"
-	"io"
 )
 
 // Saver saves module metadata and its source to underlying storage
 type Saver interface {
-	Save(ctx context.Context, module, version string, mod []byte, zip io.Reader, info []byte) error
+	Save(ctx context.Context, module, version string, mod []byte, zip Zip, info []byte) error
```

Comment:
Same here, should also be a pointer to Zip so that people are aware they're "sharing" a Zip struct and not copying it.